include/chip2chip_base.h

```diff
@@ -0,0 +1,202 @@
+#define PAGE_BIT 				0
+#define CMD_READY_BIT 			40
+#define CMD_READY_MASK 			0x0000010000000000
+
+/* acknowledge*/
+#define ACK_ERASE_ERR 			0
+#define ACK_ERASE_DONE 			1
+#define ACK_WRITE_DONE 			2
+#define ACK_BIT 			9
+#define ACK_TAG_BIT 			11
+#define ACK_CODE_MASK 			0x0000000000000600
+#define ACK_TAG_MASK 			0x000000000003f800
+#define WR_ER_ACK_READY_MASK 		0x0000000000040000
+#define WR_ER_ACK_VALID	 		0x0000000000080000
+
+#define VALID 				1
+#define INVALID 			0
+
+/* write*/
+#define WR_DATA_TAG_BIT 		20
+#define WR_DATA_TAG_VALID_BIT 	27
+#define WAIT_WRDATA_READY_BIT 	41
+#define WR_DATA_READY_MASK 		0x0000020000000000
+#define WR_DATA_REQ_READY_MASK 	0x0000000000000080
+#define WR_DATA_REQ_TAG_MASK 	0x000000000000007f
+#define WR_DATA_REQ_VALID	 	0x0000000000000100
+#define WR_DATA_REQ_TAG_BIT 	0
+/* read */
+#define READQ_READY_MASK 		0x8080808080808080
+#define READQ_TAG_MASK 			0x7f7f7f7f7f7f7f7f
+#define READ_DATA0_READY_VALUE 	0x0000000000000080
+#define READ_DATA1_READY_VALUE 	0x0000000000008000
+#define READ_DATA2_READY_VALUE 	0x0000000000800000
+#define READ_DATA3_READY_VALUE 	0x0000000080000000
+#define READ_DATA4_READY_VALUE 	0x0000008000000000
+#define READ_DATA5_READY_VALUE 	0x0000800000000000
+#define READ_DATA6_READY_VALUE 	0x0080000000000000
+#define READ_DATA7_READY_VALUE 	0x8000000000000000
+#define READ_DATA0_TAG_MASK		0x000000000000007f
+#define READ_DATA1_TAG_MASK		0x0000000000007f00
+#define READ_DATA2_TAG_MASK		0x00000000007f0000
+#define READ_DATA3_TAG_MASK		0x000000007f000000
+#define READ_DATA4_TAG_MASK		0x0000007f00000000
+#define READ_DATA5_TAG_MASK		0x00007f0000000000
+#define READ_DATA6_TAG_MASK		0x007f000000000000
+#define READ_DATA7_TAG_MASK		0x7f00000000000000
+#define READ_DATA0_TAG_BIT 		0
+#define READ_DATA1_TAG_BIT 		8
+#define READ_DATA2_TAG_BIT 		16
+#define READ_DATA3_TAG_BIT 		24
+#define READ_DATA4_TAG_BIT 		32
+#define READ_DATA5_TAG_BIT 		40
+#define READ_DATA6_TAG_BIT 		48
+#define READ_DATA7_TAG_BIT 		56
+
+/* max waiting time*/
+#define MAX_WR_DATA_REQ_WAIT_CNT 	100
+#define MAX_READ_STATUS_WAIT_CNT 	100
+#define MAX_WRITE_STATUS_WAIT_CNT 	10000
+#define MAX_ERASE_STATUS_WAIT_CNT 	100000
+#define MAX_CMD_RDY_WAIT_CNT 		10000
+#define MAX_WR_DATA_READY_CNT 		10000
+
+#define REG_SIZE			0xA8
+
+
+typedef unsigned long long u64;
+
+struct _rgstr_vptr {
+	u64 *cmd;		//Command register
+	u64 *read_stat;		//Read status register
+	u64 *read_data_0u;	//Read data upper register
+	u64 *read_data_0l;	//Read data lower register
+	u64 *read_data_1u;
+	u64 *read_data_1l;
+	u64 *read_data_2u;
+	u64 *read_data_2l;
+	u64 *read_data_3u;
+	u64 *read_data_3l;
+	u64 *read_data_4u;
+	u64 *read_data_4l;
+	u64 *read_data_5u;
+	u64 *read_data_5l;
+	u64 *read_data_6u;
+	u64 *read_data_6l;
+	u64 *read_data_7u;
+	u64 *read_data_7l;
+	u64 *wne_stat;		//Write and erase status register
+
+	u64 *write_data_u;	//Write data upper register
+	u64 *write_data_l;	//Write data lower register
+};

```

Comment:
```c
#include <stdio.h>
#include <inttypes.h>

struct address {
  union {
    struct {
      uint64_t page:16;
      uint64_t block:16;
      uint64_t chip:16;
      uint64_t bus:16;
    } format;
    uint64_t cmd;
  };
};

int main(void) {
  struct address addr;
  addr.cmd = 0;
  addr.format.page = 1;
  addr.format.block = 2;
  addr.format.chip = 1;
  addr.format.bus = 16;
  printf("0x%lx\n", addr.cmd);
  return 0;
}
```

Reply:
I aim to do this.

![image](https://github.com/BlaCkinkGJ/Flash-Translation-Layer/assets/16631264/5e79a671-0e54-4e99-9b5a-6e63d7a39ae7)